Stratis.Bitcoin/BlockPulling/BlockPullerBehavior.cs

```diff
@@ -121,13 +144,20 @@ private void Node_MessageReceived(Node node, IncomingMessage message)
         /// </summary>
         internal void AssignPendingVector()
         {
+            this.logger.LogTrace($"[{this.GetHashCode():x}] ()");
+
             Node attachedNode = this.AttachedNode;
             if (attachedNode == null || attachedNode.State != NodeState.HandShaked || !this.puller.Requirements.Check(attachedNode.PeerVersion))
+            {
+                this.logger.LogTrace($"[{this.GetHashCode():x}] (-)[ATTACHED_NODE]");
```

Comment:
it is just to distinguish different exit from the method, `(-)` says we are existing the method and `[XYZ]` identifies which exit this is (standard exit is without any id), so reading the log seeing this line, you will know that the execution flow exited here because of this if